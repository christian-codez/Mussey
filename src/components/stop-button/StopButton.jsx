import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSquare } from '@fortawesome/free-solid-svg-icons';
import { stopMusic } from '../../redux/actions/playerActions';
import { connect } from 'react-redux';
import { stopTrack } from '../../utils/songUtils';

const StopButton = ({ song, audioRef, stopMusic }) => {
  const stopSong = () => {
    stopMusic();
    stopTrack(audioRef);
  };

  return (
    <div
      onClick={() => stopSong()}
      className={` ${!song ? 'disabled' : ''} repeat-song pointer-cursor`}>
      <FontAwesomeIcon icon={faSquare} />
    </div>
  );
};

export default connect(null, { stopMusic })(StopButton);
