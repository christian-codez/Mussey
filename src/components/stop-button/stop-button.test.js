import React from 'react';
import { shallow, render, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import StopButton from './StopButton';

const mockStore = configureMockStore();
const store = mockStore({});

it('should render without throwing an error', () => {
  expect(
    shallow(
      <Provider store={store}>
        <StopButton />
      </Provider>
    ).length
  ).toEqual(1);

  expect(
    shallow(
      <Provider store={store}>
        <StopButton />
      </Provider>
    )
  ).toMatchSnapshot();
});
