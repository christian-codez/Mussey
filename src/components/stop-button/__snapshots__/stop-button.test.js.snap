// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `ShallowWrapper {}`;
